### DISCUSSION TITLE
Select filter on startup

### DISCUSSION THREAD

**Discussion by PJGaetan** [charmbracelet/huh#562] [Ideas]
I am looking into `Select`, and right now by default, filtering is false on startup.

```go
return &Select[T]{
	options:   []Option[T]{},
	value:     new(T),
	validate:  func(T) error { return nil },
	filtering: false,
	filter:    filter,
	theme:     ThemeCharm(),
}
```

I'm trying to get a user experience where the select is a _filter_ mode when `Run` starts.

I wonder if there is a way to do so already, but I did not find how to override the `filtering` parameter before rendering the prompt. 

I have in mind something like this, to a stay in line with what exist already.

```go
func (s *Select[T]) Filtering(filtering bool) *Select[T] {
	s.filtering = filtering
	return s
}
```

Happy to contribute, if this is something you would welcome.

**Comment by maaslalani** [comment]
Hey! @PJGaetan This makes sense to me! We'll just need to make sure the `s.filter` textinput gets focused as well, this should be done on `Update` if the `s.filtering` is true but `s.filter.Focused` is false.

**Comment by bashbunni** [comment]
Hey @PJGaetan, thank you so much for wanting to make this project better! I've converted this to a discussion so we can talk through it and keep our backlog clear. If it does end up being something we want to add, we'll promote it back to an issue when we are ready to work on it.

If we commit to this feature, that's a feature we will need to maintain for the lifespan of this project. So, since "Yes is forever", we are carefully considering new features.

**Comment by PJGaetan** [comment]
Hey @bashbunni, thank you for looking into this.

The issue was never closed but an MR was accepted last year to handle this issue.
https://github.com/charmbracelet/huh/pull/179

I think the discussion can be closed as well ! :)
